daemon/cmd/daemon_main.go

```diff
@@ -1918,6 +1919,45 @@ func startDaemon(d *Daemon, restoredEndpoints *endpointRestoreState, cleaner *da
 	if err != nil {
 		log.WithError(err).Error("Unable to store Viper's configuration")
 	}
+
+	const (
+		epsBPFSync         = "eps-bpf-sync"
+		epsBPFSyncDuration = 30 * time.Second
+	)
+	controller.NewManager().UpdateController(
+		epsBPFSync,
+		controller.ControllerParams{
+			Group: controller.NewGroup(epsBPFSync),
+			DoFunc: func(ctx context.Context) error {
+				var (
+					reload bool
+					eps    = d.endpointManager.GetEndpoints()
+				)
+				// !!BOZO!! Debug...
+				defer func(t time.Time) {
+					log.Debugf("Endpoints BPF prog reload elapsed [%t]: %s -- (%d)", reload, time.Since(t), len(eps))
+				}(time.Now())
+
+				for _, ep := range eps {
+					reload, err = endpoints.NeedsBPFProgReload(log, endpoints.NewEndpointBPFProgBacker(ep))
+					if err != nil {
+						return err
+					}
+					if reload {
+						break
+					}
+				}
+				if !reload {
+					return nil
+				}
+
+				log.Debugf("Reloading endpoint BPF programs (%d)", len(eps))
+				_, err := d.TriggerReloadWithoutCompile(epsBPFSync)
+				return err
+			},
+			RunInterval: epsBPFSyncDuration,
+			Context:     d.ctx,
+		})
```

Comment:
what do you think about putting this whole chunk into its own function, rather than directly in startDaemon?

Reply:
Good point! I'll make it so.